lib/msal-browser/docs/errors.md

```diff
@@ -142,6 +143,66 @@ Remember that you will need to register this new `redirectUri` on your App Regis
+
+✔️ To solve this problem you should ensure that the page you use as your `redirectUri` is not doing any of these things, at the very least, when loaded in a popup or iframe. We recommend using a blank page as your `redirectUri` for silent and popup flows to ensure none of these things can occur.
+
+You can do this on a per request basis, for example:
+
+```javascript
+msalInstance.acquireTokenSilent({
+    scopes: ["User.Read"],
+    redirectUri: "http://localhost:3000/blank.html"
+});
+```
+
+Remember that you will need to register this new `redirectUri` on your App Registration.
+
+**Notes regarding Angular and React:**
+
+- If you are using `@azure/msal-angular` your `redirectUri` page should not be protected by the `MsalGuard`.
+- If you are using `@azure/msal-react` your `redirectUri` page should not render the `MsalAuthenticationComponent` or use the `useMsalAuthentication` hook.
+
+#### Issues caused by the Identity Provider
+
+You can also get this error if the Identity Provider fails to redirect back to your application. In silent scenarios this error is sometimes accompanied by an X-Frame-Options: Deny error indicating that your identity provider is attempting to either show you an error message or is expecting interaction. The X-Frame-Options error will usually have a url in it and opening this url in a new tab may help you discern what is happening.
+
+Some B2C flows are expected to throw this error due to their need for user interaction. These flows include:
+
+- Password reset
+- Profile edit
+- Sign up
+- Some custom policies depending on how they are configured
+
+Another potential reason the identity provider may not redirect back to your application in time may be that there is some extra network latency. The default timeout is about 10 seconds and should be sufficient in most cases, however, if your identity provider is taking longer than that to redirect you can increase this timeout in the MSAL config with either the `iframeHashTimeout`, `windowHashTimeout` or `loadFrameTimeout` configuration parameters.
+
+```javascript
+const msalConfig = {
+    auth: {
+        clientId: "your-client-id"
+    },
+    system: {
+        windowHashTimeout: 9000, // Applies just to popup calls - In milliseconds
+        iframeHashTimeout: 9000, // Applies just to silent calls - In milliseconds
+        loadFrameTimeout: 9000 // Applies to both silent and popup calls - In milliseconds
+    }
+};
+```
+
```

Comment:
For AAD, I think it there should be a section talking about throttling, as it is the most common cause of this error. I think we should also talk about mitigations, which can include:

1. Retrying the request and/or
2. Invoking interaction

Reply:
Done